### PR TITLE
Let legiond replace fancurve-set completely

### PR DESCRIPTION
**Pull request by st0nie**
CC @MrDuartePT 

fancurve-set is too slow, I replaced its functionality with the legiond module.

Note: User need to migrate their `/etc/legiond/.env` to `/etc/legiond/legiond.ini`
`legiond-cli` rename to `legiond-ctl` to prevent confusion with legion_cli

### CODE REVIEW

**Comment by MrDuartePT** [comment]
Pls Rebase `fix typo` commit

**Comment by MrDuartePT** [comment]
@johnfanv2 i think would be a good ideia to add @st0nie as a mantainer

**Comment by st0nie** [comment]
> Pls Rebase `fix typo` commit

ok

**Comment by MrDuartePT** [comment]
Well if all check pass I will test it and then accept it

**Comment by MrDuartePT** [comment]
The fail is not your code problem is my dockerfile is broken

**Comment by st0nie** [comment]
> The fail is not your code problem is my dockerfile is broken

Oh I forget fclose, let me fix it

**Comment by MrDuartePT** [comment]
> > The fail is not your code problem is my dockerfile is broken
> 
> Oh I forget fclose, let me fix it

This would require changes to the openrc files. Do me a favor add them to the repo to the service folder.

**Comment by st0nie** [comment]
> This would require changes to the openrc files. I gonna add them to the service folder

remember to add inih to build dependencies
I use this lib to parse ini

https://github.com/benhoyt/inih/blob/master/ini.h

**Comment by MrDuartePT** [comment]
Thanks for the heads up

**Comment by MrDuartePT** [comment]
Funny your code not likes lto:
```
 *   Installing legion_linux-9999-py3-none-any.whl to /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/work/lenovolegionlinux-9999-python3_12/install
python3.12 -m gpep517 install-wheel --destdir=/var/tmp/portage/sys-firmware/lenovolegionlinux-9999/work/lenovolegionlinux-9999-python3_12/install --interpreter=/usr/bin/python3.12 --prefix=/usr --optimize=all /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/work/lenovolegionlinux-9999-python3_12/wheel/legion_linux-9999-py3-none-any.whl
2024-04-05 13:13:07,349 gpep517 INFO Installing /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/work/lenovolegionlinux-9999-python3_12/wheel/legion_linux-9999-py3-none-any.whl into /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/work/lenovolegionlinux-9999-python3_12/install
2024-04-05 13:13:07,423 gpep517 INFO Installation complete
make -j14 -l14
clang -march=native -O3 -pipe -fdiagnostics-color=always -flto=thin -o legiond legiond.c modules/parseconf.c modules/powerstate.c modules/setapply.c
clang -march=native -O3 -pipe -fdiagnostics-color=always -flto=thin -o legiond-ctl legiond-ctl.c
ld.lld: error: undefined symbol: ini_parse
>>> referenced by parseconf.c
>>>               /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/temp/legiond.lto.parseconf-835772.o:(parseconf)
>>> referenced by legiond.c
>>>               /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/temp/legiond.lto.legiond-00ccc0.o:(main)
>>> referenced by legiond.c
>>>               /var/tmp/portage/sys-firmware/lenovolegionlinux-9999/temp/legiond.lto.legiond-00ccc0.o:(main)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:11: legiond] Error 1
 * ERROR: sys-firmware/lenovolegionlinux-9999::guru failed (compile phase):
 *   emake failed
 ```

**Comment by st0nie** [comment]
wait I need do some change

**Review by MrDuartePT — COMMENTED** [review]
@st0nie why this commit is touching the legion-gui.py

**Comment by MrDuartePT** [comment]
@st0nie you basically remove all my changes I will go have to force push, you forget to rebase before making changes

**Comment by st0nie** [comment]
> @st0nie you basically remove all my changes I will go have to force push, you forget to rebase before making changes

Sorry, I forget to pull first😢

**Comment by MrDuartePT** [comment]
> > @st0nie you basically remove all my changes I will go have to force push, you forget to rebase before making changes
> 
> Sorry, I forget to pull first😢

No problem I will solve this

**Comment by MrDuartePT** [comment]
@st0nie I removed the 3 commit you added but everthing should be right, you just need to add the docs

**Comment by st0nie** [comment]
> @st0nie I removed the 3 commit you added but everthing should be right, you just need to add the docs

Is `legiond.initd` wrong?
the `legiond-ctl reload` is used to reload `legiond.ini`, not the service itself

**Comment by MrDuartePT** [comment]
> > @st0nie I removed the 3 commit you added but everthing should be right, you just need to add the docs
> 
> Is `legiond.initd` wrong? the `legiond-ctl reload` is used to reload `legiond.ini`, not the service itself

Oh ok, yup it wrong you can solve it.
By the way also solve this file: https://github.com/johnfanv2/LenovoLegionLinux/pull/195/commits/64d619c3a4808ef13b907986a2787a1104de70eb#diff-8c01cd0b44a718a015412f7f115b6bb946e4d356c501b80db987dd9562752c3a
It should be this:
```bash
#!/bin/bash
while true; do
     /usr/bin/legiond-ctl cpuset
     sleep 30
done
```

**Comment by MrDuartePT** [comment]
Well I notice a file that forget to commit, but do your changes I will add that commit later

**Comment by MrDuartePT** [comment]
Ok I gonna correct it, you need to do more changes @st0nie

**Comment by MrDuartePT** [comment]
LGTM
See if you want to add anything else.

File: extra/service/legiond-cpuset.initd [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/sbin/openrc-run
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+command="/usr/sbin/legion-cpuset"
+command_background=true
+pidfile="/run/legiond.pid"
```
Comment (st0nie) [inline extra/service/legiond-cpuset.initd:7]: pidfile="/run/legiond-cpuset.pid"

File: extra/service/legiond-onresume.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+case $1/$2 in
+post/*)
+       /usr/bin/legiond-cli fanset 3
```
Comment (st0nie) [inline extra/service/legiond-onresume.sh:4]: legiond-ctl

**Comment by st0nie** [comment]
> LGTM See if you want to add anything else.

^^^^

Find some small problems

File: deploy/build_packages/lenovolegionlinux.spec [hunk 3]
```diff
@@ -50,7 +50,7 @@ install -D -m 0644 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_l
 install -D -m 0644 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_linux/legion_linux/extra/service/legiond-onresume.service %{_unitdir}/legiond-onresume.service
 install -D -m 0644 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_linux/legion_linux/extra/service/legiond.service %{_unitdir}/legiond-cpuset.service
 install -D -m 0644 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_linux/legion_linux/extra/service/legiond.service %{_unitdir}/legiond-cpuset.timer
-install -D -m 0755 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_linux/legion_linux/extra/service/legiond/legiond-cli %{_bindir}/legiond_cli
+install -D -m 0755 %{_builddir}/%{srcname}-%{version}-prerelease/python/legion_linux/legion_linux/extra/service/legiond/legiond-ctl %{_bindir}/legiond_ctl
```
[the remark this answers is not in the document]
Comment (MrDuartePT) [reply deploy/build_packages/lenovolegionlinux.spec]: That both are solved

File: deploy/build_packages/build_deb_LLL.sh [hunk 4]
```diff
@@ -61,15 +61,15 @@ sudo cp -R ${REPODIR}/extra/service/legiond-onresume.service .
...
+sudo mv ${REPODIR}/extra/service/legiond/legiond-ctl .
 sudo cp -R ${REPODIR}/extra/acpi/events/legion_ac .
 sudo cp -R ${REPODIR}/extra/acpi/events/legion_ppd .
 echo "legiond.service /etc/systemd/system/" | sudo tee -a debian/install
 echo "legiond-onresume.service /etc/systemd/system/" | sudo tee -a debian/install
 echo "legiond-cpuset.service /etc/systemd/system/" | sudo tee -a debian/install
 echo "legiond-cpuset.timer /etc/systemd/system/" | sudo tee -a debian/install
 echo "legiond /usr/bin/legiond" | sudo tee -a debian/install
-echo "legiond-cli /usr/bin/legiond_cli" | sudo tee -a debian/install
+echo "legiond-cli /usr/bin/legiond_ctl" | sudo tee -a debian/install
```
[the remark this answers is not in the document]
Comment (MrDuartePT) [reply deploy/build_packages/build_deb_LLL.sh]: That both are solved

**Comment by st0nie** [comment]
After u fix that I wanna add some todo to the README of legiond

File: extra/service/legiond-cpuset.initd (hunk 1 above)
Comment (MrDuartePT) [reply extra/service/legiond-cpuset.initd]: thanks

File: extra/service/legiond-onresume.sh (hunk 2 above)
Comment (MrDuartePT) [reply extra/service/legiond-onresume.sh]: thanks

**Review by st0nie — COMMENTED** [review]
Nice work

**Comment by st0nie** [comment]
Emmm better use checkbox

**Comment by st0nie** [comment]
ready to merge 🚀

**Comment by MrDuartePT** [comment]
> Emmm better use checkbox

Org checkboxes seems to not work well on github, but it ok

**Comment by MrDuartePT** [comment]
@st0nie you have any ideia why ubuntu is failing to compile: https://github.com/johnfanv2/LenovoLegionLinux/actions/runs/8572175765/job/23494315069

**Comment by st0nie** [comment]
> @st0nie you have any ideia why ubuntu is failing to compile: https://github.com/johnfanv2/LenovoLegionLinux/actions/runs/8572175765/job/23494315069
> 
> It can be the old gcc version

Yeah, it might be
image

clang on Ubuntu works fine

**Pull request merged**